simpleviewinc/keg-hub
# Move helper methods from ReTheme to jsutils

## Context

* There were a few methods that should go into jsutils instead of ReTheme
* We should move them to jsutils

## Goal

* Move some methods form ReTheme to  jsutils 

## Updates
* Added a couple array helper methods
  * `flatArr` - Makes an array with arrays flat 
  * `ensureArr` - Ensures a variable is an array
* Migrated some methods from ReTheme to jsutils
  *  `hashstring` - Converts a string in to a hash
  * `hyphenator` - Converts a CSS in JS rule  string to standard CSS style with hyphens  
* Updated calls to `hasDomAccess` to use the jsutils version

## Testing

**Test 1**
* Pull the PR => `keg pr 115`
* Run tests for jsutils => `keg jsutils && yarn test`
* Run tests for ReTheme => `keg retheme && yarn test`

**Test 2**
* Run this `Re-theme` docker package => `keg retheme pack run docker.pkg.github.com/simpleviewinc/keg-packages/retheme:add-storybook-to-retheme`
  * In your browser navigate to http://retheme-add-storybook-to-retheme.local.kegdev.xyz
  * Ensure everything works as expected

**Test 3**
* Run this `Keg-Components` docker package => `keg dpg run docker.pkg.github.com/simpleviewinc/keg-packages/keg-components:add-storybook-to-retheme`
  * This has the `jsutils` and  `ReTheme` changes synced into it
*  In your browser navigate to http://components-add-storybook-to-retheme.local.kegdev.xyz/
  * Ensure **ALL** components render as expected
